src/anomalib/models/components/base/anomaly_module.py

```diff
@@ -51,6 +51,21 @@ def __init__(self) -> None:
         self.pixel_metrics: AnomalibMetricCollection
 
         self._transform: Transform | None = None
+        self._input_size: tuple[int, int] | None = None
+
+    def setup(self, stage: str) -> None:
+        """Calls the _setup method to build the model if the model is not already built."""
+        del stage
+        if getattr(self, "model", None) is None:
+            self._setup()
+
+    def _setup(self) -> None:
```

Comment:
Do we really need `_setup`? Can't we do this in setup directly?

Reply:
By having a separate `_setup` method we ensure that the checking logic (`if getattr(self, "model", None) is None:`) is always applied for all models. This is important, because if we don't apply this check, the following would have the trained model overwritten by a empty one the `test` call.

```python
>>> engine = Engine()
>>> model = Padim()
>>> data = MVTec()
>>>
>>> engine.fit(model, datamodule=data)
>>> engine.test(model, datamodule=data)
```

Reply:
Ah right! People might miss the `if getattr(self, "model", None) is None:` line in their implementation